src/cmd-koji-upload

```diff
@@ -682,33 +683,58 @@ class Upload(_KojiBase):
         """
         serverdir = klib.unique_path(f"{self.build.build_id}-cosa")
         log.debug(f"uploading files to {serverdir}")
+        meta_path = ""
+        meta_remote_path = ""
+        callback = klib._progress_callback
+
         for _, meta in (self.build).get_artifacts():
             local_path = meta['local_path']    # the local file to upload
             remote_path = meta['upload_path']  # the name of the file to upload
-            log.info(f"Uploading {local_path} to {serverdir}/{remote_path}")
-            callback = klib._progress_callback
+            if ("meta.json" in local_path) and ("commitmeta.json" not in local_path):
+                # Keep location for meta.json
+                log.info(f"Uploading {local_path} to {serverdir}/{remote_path}")
+                meta_path = local_path
+                meta_remote_path = remote_path
             self.session.uploadWrapper(local_path, serverdir, name=remote_path,
                                        callback=callback)
             if callback:
                 print('')
 
         self._uploaded = True
         self._remote_directory = serverdir
+        cginfo = ""
 
-        """ If the final Import fails retry it """
-        for attempt in range(0, self._retry_attempts):
+        # If the final Import fails retry it
+        for attempt in range(1, self._retry_attempts + 1):
+            self.manifest
             try:
+                log.info(f"Finishing Koji Upload - Try {attempt}/{self._retry_attempts}")
                 cginfo = self.session.CGImport(self.manifest, serverdir, token=self._token)
+                log.info(json.dumps(cginfo, sort_keys=True, indent=3))
+                log.info(f"Recorded build {cginfo['nvr']}")
+                # Koji has some flakes here, confirm the nvr to make sure the upload worked
+                if cginfo['nvr']:
+                    break
```

Comment:
If Koji is flaky around here, should we check `cginfo['nvr']` first before we dump log messages about the recorded build?

```suggestion
                # Koji has some flakes here, confirm the nvr to make sure the upload worked
                if cginfo['nvr']:
                    log.info(json.dumps(cginfo, sort_keys=True, indent=3))
                    log.info(f"Recorded build {cginfo['nvr']}")
                    break
```